includes/forms.inc

```diff
@@ -358,6 +358,38 @@ function islandora_ip_embargo_add_range_form($form, &$form_state, $list_identifi
 }
 
 /**
+ * Validate handler for adding an IP Range.
+ *
+ * @param array $form
+ *   The Drupal form definition.
+ * @param array $form_state
+ *   The Drupal form state.
+ */
+function islandora_ip_embargo_add_range_form_validate(array $form, array &$form_state) {
+  $comparable_low = ip2long($form_state['values']['low_end']);
+  $comparable_high = ip2long($form_state['values']['high_end']);
+
+  $error = FALSE;
+  if (empty($comparable_low)) {
+    $error = TRUE;
+    form_set_error('low_end', t('Low end IP is not valid.'));
+  }
+
+  if (empty($comparable_high)) {
+    $error = TRUE;
+    form_set_error('high_end', t('High end IP is not valid.'));
+  }
+
+  if ($error) {
+    form_set_error('', t('IP addresses must be in dot-decimal notation with no leading zeros.'));
```

Comment:
can you give me an example of what should work? I've tried a couple of different formats, ip2long hasn't appreciated any of them yet.

0xC0.0x00.0x02.0xEB, 0300.0000.0002.0353, 0xC00002EB, 3221226219, and 030000001353 do not work in ip2long